h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java

```diff
@@ -203,14 +206,33 @@ protected int desiredChunks(final Frame original_fr, boolean local) {
   public class DeepLearningDriver extends Driver {
     @Override public void computeImpl() {
       init(true); //this can change the seed if it was set to -1
-      long cs = _parms.checksum();
+      if (_parms._evaluate_auto) {
+        initActualParamValues();
+      }
+      Model.Parameters parmsToCheck = _parms.clone();
       // Something goes wrong
       if (error_count() > 0)
         throw H2OModelBuilderIllegalArgumentException.makeFromBuilder(DeepLearning.this);
       buildModel();
       //check that _parms isn't changed during DL model training
-      long cs2 = _parms.checksum();
-      assert(cs == cs2);
+      Model.Parameters parmsToCheck2 = _parms.clone();
```

Comment:
why do we need this clone, does checkNonAutoParmsNotChanged modify any of its arguments?

Reply:
before, it was checking all the parameters to not to be changed. Now I need to evaluate/change distribution parameter in case it is AUTO to its actual value during buildModel(), because I can't tell it beforehand, it depends on nClasses and loss function and I set it after the place where it is evaluated and set to model_info inside buildModel() inside DeepLearningParameters.Sanity.modifyParms(_parms, cp.model_info().get_params(), nclasses()); (DeepLearning.javaL312)
Hence I rewrote this check: it assumes that parameters with value AUTO will be evaluated during buildModel() (which is only distribution parameter) and checks for not being changed all the others (instead checking all the parameters).